community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/Operations.java

```diff
@@ -323,11 +323,6 @@ public CapableIndexReference index( int label, int... properties )
 
     private void assertOpen()
     {
-        if ( !ktx.isOpen() )
-        {
-            throw new NotInTransactionException( "The transaction has been closed." );
-        }
```

Comment:
Looks like a reasonable thing to do, though.

Reply:
Yes, if I thought so to but `ktx.isOpen`  is `true` when `closing` is true which is when transaction hooks run and they expect to be able to use the transaction anyway. I would love to clear up around `isOpen`, `assertOpen` et. al because they all do slightly different stuff.  In this way we - at least to my knowledge - we do the same as we did before.